common/model-views.cpp

```diff
@@ -2617,77 +2617,177 @@ namespace rs2
+            std::string units;
+            std::string toolTip;
+            int nameExtraSpace;
+        };
+
+        std::vector<pose_data> pose_vector = {
+            { "Velocity", {pose_frame.velocity.x, pose_frame.velocity.y , pose_frame.velocity.z , FLT_MAX }, "(Meter/Sec)", "Velocity: X, Y, Z values of velocity, in Meter/Sec", 0},
+            { "Angular Velocity",{ pose_frame.angular_velocity.x, pose_frame.angular_velocity.y , pose_frame.angular_velocity.z , FLT_MAX }, "(Radians/Sec)", "Angular Velocity: X, Y, Z values of angular velocity, in Radians/Sec", 0 },
+            { "Acceleration",{ pose_frame.acceleration.x, pose_frame.acceleration.y , pose_frame.acceleration.z , FLT_MAX }, "(Meter/Sec^2)", "Acceleration: X, Y, Z values of acceleration, in Meter/Sec^2", 0 },
+            { "Angular Acceleration",{ pose_frame.angular_acceleration.x, pose_frame.angular_acceleration.y , pose_frame.angular_acceleration.z , FLT_MAX }, "(Radians/Sec^2)", "Angular Acceleration: X, Y, Z values of angular acceleration, in Radians/Sec^2", 0 },
+            { "Translation",{ pose_frame.translation.x, pose_frame.translation.y , pose_frame.translation.z , FLT_MAX }, "(Meter)", "Translation: X, Y, Z values of translation in Meter (relative to initial position)", 0 },
+            { "Rotation",{ pose_frame.rotation.x, pose_frame.rotation.y , pose_frame.rotation.z , pose_frame.rotation.w }, "", "Rotation: Qi, Qj, Qk, Qr components of rotation as represented in quaternion rotation (relative to initial position)", 0 },
+        };
+
+        int line_h = 18;
+        for (auto&& pose : pose_vector)
+        {
+            auto rc = ImGui::GetCursorPos();
+            ImGui::SetCursorPos({ rc.x + 12, rc.y + 4 });
+            ImGui::Text("%s:", pose.name.c_str());
+            if (ImGui::IsItemHovered())
+            {
+                ImGui::SetTooltip(pose.toolTip.c_str());
             }
 
-            ImGui::End();
-            ImGui::PopStyleColor(6);
-            ImGui::PopStyleVar(2);
+            switch (pose_frame.tracker_confidence) //color the line according to confidence
+            {
+            case 3: // High confidence - Green
+                ImGui::PushStyleColor(ImGuiCol_Text, green);
+                break;
+            case 2: // Medium confidence - Yellow
+                ImGui::PushStyleColor(ImGuiCol_Text, yellow);
+                break;
+            case 1: // Low confidence - Red
+                ImGui::PushStyleColor(ImGuiCol_Text, red);
+                break;
+            case 0: // failed confidence - Grey
+                ImGui::PushStyleColor(ImGuiCol_Text, grey);
+                break;
+            }
+
+            ImGui::SetCursorPos({ rc.x + 150 + pose.nameExtraSpace, rc.y + 1 });
+            std::string label = to_string() << "##" << pose.name.c_str();
+
+            std::string data = "[";
+            std::string comma = "";
+            unsigned int i = 0;
+            while ((pose.data[i] != FLT_MAX) && (i<4))
```

Comment:
Is the extra-verbosity by design ? If so - please put comments.

Reply:
non issue